### PR TITLE
Initial AIE profiling additions

### PR DESCRIPTION
**Pull request by pgschuey**
* Addition of shim/PL profiling
* New metric sets for core and memory modules
* Appropriate configuration of stream switch ports for monitoring

### CODE REVIEW

**Review by stsoe — APPROVED** [review]

**Review by rozumx — APPROVED** [review]
Several comments around formatting, but nothing major.  Approving PR.

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp [hunk 1]
```diff
@@ -258,38 +281,39 @@ namespace xdp {
...
+          : ((mod == XAIE_MEM_MOD) ? mMemoryEventStrings[metricSet] : mShimEventStrings[metricSet]);
 
     auto numTotalEvents = requestedEvents.size();
     if (numFreeCtr < numTotalEvents) {
       std::stringstream msg;
-      std::string modType = isCore ? "core" : "memory";
       msg << "Only " << numFreeCtr << " out of " << numTotalEvents
-          << " metrics were available for aie "
-          << modType <<  " module profiling due to resource constraints. "
+          << " metrics were available for AIE "
```
[the remark this answers is not in the document]
Comment (pgschuey) [reply src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp:296]: module names are simple so not needed

**Comment by pgschuey** [comment]
please re-test

**Comment by dayeh-xilinx** [comment]
retest this please. Failed on faulty u250 test.

**Comment by pgschuey** [comment]
@hackwa

**Review by jvillarre — APPROVED** [review]

**Pull request merged**